### ISSUE TITLE
load8 vs load_word?

### ISSUE THREAD

**Issue by andres-erbsen** [mit-plv/bedrock2#159]
If we can have multiplication in the AST and still get compiler proofs for rv32im (if the code does not use it), can we have load8 in the AST and still get compiler proof for 32-bit architectures (if the code does not use load8). This would make it easier for the C back-end to satisfy strict aliasing rule (https://github.com/mit-plv/fiat-crypto/issues/821#issuecomment-649681246).

**Comment by samuelgruetter** [comment]
Aaww, do you remember when we defined
https://github.com/mit-plv/bedrock2/blob/b28c1dc5e2c35238acb05cd0aed67176afaa71b6/bedrock2/src/bedrock2/Syntax.v#L11
how smart we felt that this automatically allows 8-byte load/store only on 64bit architectures? :wink:

Is this not sufficient any more? As in, why couldn't you use `access_size.word` for load8?

If needed, you could probably change `access_size` to
```
  Variant access_size := one | two | four | word | eight.
```
but before doing that, I'd like to properly understand why it's needed.

**Comment by andres-erbsen** [comment]
In the idiomatic C translation of these constructs, storing using "store8" and then loading using "loadword" is a strict aliasing violation. And the code that would be calling bedrock2-printed-as-C sure uses `uint64_t` not `uintptr_t`.

**Comment by andres-erbsen** [comment]
Another thought: we could just play safe and stupid with strict aliasing and load larger words in 8-bit chunks...

**Comment by samuelgruetter** [comment]
Could you write bitwidth-specific bedrock2-to-C pretty-printers that never output `uintptr_t`, but only `uint32/64_t`?

**Comment by andres-erbsen** [comment]
Yes, we could. Using the wrong one could lead to really awkward bugs.
